src/main/kotlin/io/snyk/plugin/snykcode/SnykCodeBulkFileListener.kt

```diff
@@ -28,50 +29,71 @@ class SnykCodeBulkFileListener : SnykBulkFileListener() {
         CacheBuilder.newBuilder()
             .expireAfterWrite(Duration.ofMillis(1000)).build<String, Boolean>()
 
+    private val blackListedDirectories =
+        setOf(".idea", ".git", ".hg", ".svn")
+
     override fun before(project: Project, virtualFilesAffected: Set<VirtualFile>) = Unit
 
     override fun after(project: Project, virtualFilesAffected: Set<VirtualFile>) {
         if (virtualFilesAffected.isEmpty()) return
-        ProgressManager.getInstance().run(object : Backgroundable(
-            project,
-            "Snyk: forwarding save event to Language Server"
-        ) {
-            override fun run(indicator: ProgressIndicator) {
-                val languageServerWrapper = LanguageServerWrapper.getInstance()
-                if (!languageServerWrapper.isInitialized) return
-                val languageServer = languageServerWrapper.languageServer
-                val cache = getSnykCachedResults(project)?.currentSnykCodeResultsLS ?: return
-                val filesAffected = toSnykFileSet(project, virtualFilesAffected)
-                for (file in filesAffected) {
-                    val virtualFile = file.virtualFile
-                    if (!shouldProcess(virtualFile)) continue
-                    cache.remove(file)
-                    val param =
-                        DidSaveTextDocumentParams(
-                            TextDocumentIdentifier(virtualFile.toLanguageServerURL()),
-                            virtualFile.readText()
-                        )
-                    languageServer.textDocumentService.didSave(param)
-                }
 
-                VirtualFileManager.getInstance().asyncRefresh()
-                runInEdt { DaemonCodeAnalyzer.getInstance(project).restart() }
-            }
-        })
+        runAsync {
+            if (ApplicationManager.getApplication().isDisposed) return@runAsync
+            if (project.isDisposed) return@runAsync
+            if (DumbService.getInstance(project).isDumb) return@runAsync
```

Comment:
If the project isn't indexed yet, we don't want to scan, as we may not have all workspace folders set up yet.